zcash_client_backend/src/data_api.rs

```diff
@@ -155,8 +155,10 @@ pub trait WalletRead {
     /// Returns the memo for a note.
     ///
     /// Implementations of this method must return an error if the note identifier
-    /// does not appear in the backing data store.
-    fn get_memo(&self, id_note: Self::NoteRef) -> Result<Memo, Self::Error>;
+    /// does not appear in the backing data store. Returns `Ok(None)` if the note
+    /// is known to the wallet but memo data has not yet been populated for that
+    /// note.
+    fn get_memo(&self, id_note: Self::NoteRef) -> Result<Option<Memo>, Self::Error>;
```

Comment:
Change needs to be documented in the changelog.

Reply:
Fixed in https://github.com/zcash/librustzcash/pull/843/commits/ea84243e0f4f61b997ca79c152c55824a7ded0d8